mlir_graphblas/tests/test_mlir_builder.py

```diff
@@ -0,0 +1,198 @@
+from mlir_graphblas.sparse_utils import MLIRSparseTensor
+from mlir_graphblas.mlir_builder import MLIRBuilderVar, MLIRFunctionBuilder
+from mlir_graphblas.functions import (
+    Transpose,
+    MatrixSelect,
+    MatrixReduceToScalar,
+    MatrixMultiply,
+    create_empty_matrix,
+)
+
+
+@pytest.fixture(scope="module")
+def engine():
+    jit_engine = MlirJitEngine()
+
+    jit_engine.add(
+        """
+#trait_densify = {
+  indexing_maps = [
+    affine_map<(i,j) -> (i,j)>,
+    affine_map<(i,j) -> (i,j)>
+  ],
+  iterator_types = ["parallel", "parallel"],
+  sparse = [
+    [ "D", "S" ],
+    [ "D", "D" ]
+  ],
+  sparse_dim_map = [
+    affine_map<(i,j) -> (j,i)>,
+    affine_map<(i,j) -> (i,j)>
+  ]
+}
+
+!SparseTensor = type !llvm.ptr<i8>
+
+func @densify(%argA: !SparseTensor) -> tensor<8x8xf64> {
+  %output_storage = constant dense<0.0> : tensor<8x8xf64>
+  %arga = linalg.sparse_tensor %argA : !SparseTensor to tensor<8x8xf64>
+  %0 = linalg.generic #trait_densify
+    ins(%arga: tensor<8x8xf64>)
+    outs(%output_storage: tensor<8x8xf64>) {
+      ^bb(%A: f64, %x: f64):
+        linalg.yield %A : f64
+    } -> tensor<8x8xf64>
+  return %0 : tensor<8x8xf64>
+}
+""",
+        [
+            "--test-sparsification=lower",
+            "--linalg-bufferize",
+            "--convert-scf-to-std",
+            "--func-bufferize",
+            "--tensor-bufferize",
+            "--tensor-constant-bufferize",
+            "--finalizing-bufferize",
+            "--convert-linalg-to-loops",
+            "--convert-scf-to-std",
+            "--convert-std-to-llvm",
+        ],
+    )
+    return jit_engine
+
+
+def test_ir_builder_transpose_wrapper(engine: MlirJitEngine):
+    # Build Function
+    transpose_function = Transpose()
+
+    input_var = MLIRBuilderVar("input_tensor", "!llvm.ptr<i8>")
+    output_var = MLIRBuilderVar("output_tensor", "!llvm.ptr<i8>")
+
+    ir_builder = MLIRFunctionBuilder(
+        "transpose_wrapper", output_var, input_var, return_type="index"
+    )
+    transpose_result = ir_builder.add_function_call(
+        transpose_function, output_var, input_var
+    )
+    ir_builder.add_return_statement(transpose_result)
+
+    assert ir_builder.get_mlir()
+
+    # Test Compiled Function
+    transpose_wrapper_callable = ir_builder.compile(engine=engine)
+
+    indices = np.array(
+        [
+            [1, 2],
+            [4, 3],
+        ],
+        dtype=np.uint64,
+    )
+    values = np.array([1.2, 4.3], dtype=np.float64)
+    sizes = np.array([8, 8], dtype=np.uint64)
+    sparsity = np.array([False, True], dtype=np.bool8)
+
+    input_tensor = MLIRSparseTensor(indices, values, sizes, sparsity)
+    output_tensor = create_empty_matrix(
+        *input_tensor.shape, nnz=input_tensor.pointers[1][-1]
+    )
+
+    dense_input_tensor = np.zeros([8, 8], dtype=np.float64)
+    dense_input_tensor[1, 2] = 1.2
+    dense_input_tensor[4, 3] = 4.3
+    assert np.isclose(dense_input_tensor, engine.densify(input_tensor)).all()
```

Comment:
This is redundant with the statement two lines down

Reply:
I can remove this line if it's unnecessary. My initial motivation was to make sure that the MLIR function didn't accidentally change the contents of `input_tensor`. That might be excessive.

Reply:
Ah, makes sense. I didn't understand why the same test was being run twice, but checking for no mutations is valid. Just add a comment on both lines explaining what is being tested.

Reply:
https://github.com/metagraph-dev/mlir-graphblas/pull/44/commits/9b55b48e23697775fab554e21c157ec7012de6dd#diff-8a0c1e6dbb5e24631939b88ee63968ef6e90bdad26c8539d2e0af6c193af2864R112 adds the comment.